signaling.go

```diff
@@ -0,0 +1,407 @@
+	RTCSignalingStateHaveRemoteOffer
+
+	// RTCSignalingStateHaveLocalPranswer indicates A remote description of type "offer" has been successfully applied and a local description of type "pranswer" has been successfully applied.
+	RTCSignalingStateHaveLocalPranswer
+
+	// RTCSignalingStateHaveRemotePranswer indicates A local description of type "offer" has been successfully applied and a remote description of type "pranswer" has been successfully applied.
+	RTCSignalingStateHaveRemotePranswer
+
+	// RTCSignalingStateClosed indicates The RTCPeerConnection has been closed.
+	RTCSignalingStateClosed
+)
+
+func (t RTCSignalingState) String() string {
+	switch t {
+	case RTCSignalingStateStable:
+		return "stable"
+	case RTCSignalingStateHaveLocalOffer:
+		return "have-local-offer"
+	case RTCSignalingStateHaveRemoteOffer:
+		return "have-remote-offer"
+	case RTCSignalingStateHaveLocalPranswer:
+		return "have-local-pranswer"
+	case RTCSignalingStateHaveRemotePranswer:
+		return "have-remote-pranswer"
+	case RTCSignalingStateClosed:
+		return "closed"
+	default:
+		return "Unknown"
+	}
+}
+
+// RTCSdpType describes the type of an RTCSessionDescription
+type RTCSdpType int
+
+const (
+
+	// RTCSdpTypeOffer indicates that a description MUST be treated as an SDP offer.
+	RTCSdpTypeOffer RTCSdpType = iota + 1
+
+	// RTCSdpTypePranswer indicates that a description MUST be treated as an SDP answer, but not a final answer.
+	RTCSdpTypePranswer
+
+	// RTCSdpTypeAnswer indicates that a description MUST be treated as an SDP final answer, and the offer-answer exchange MUST be considered complete.
+	RTCSdpTypeAnswer
+
+	// RTCSdpTypeRollback indicates that a description MUST be treated as canceling the current SDP negotiation and moving the SDP offer and answer back to what it was in the previous stable state.
+	RTCSdpTypeRollback
+)
+
+func (t RTCSdpType) String() string {
+	switch t {
+	case RTCSdpTypeOffer:
+		return "offer"
+	case RTCSdpTypePranswer:
+		return "pranswer"
+	case RTCSdpTypeAnswer:
+		return "answer"
+	case RTCSdpTypeRollback:
+		return "rollback"
+	default:
+		return "Unknown"
+	}
+}
+
+// RTCSessionDescription is used to expose local and remote session descriptions.
+type RTCSessionDescription struct {
+	Type RTCSdpType
+	Sdp  string
+}
+
+// SetRemoteDescription sets the SessionDescription of the remote peer
+func (r *RTCPeerConnection) SetRemoteDescription(desc RTCSessionDescription) error {
+	if r.remoteDescription != nil {
+		return errors.Errorf("remoteDescription is already defined, SetRemoteDescription can only be called once")
+	}
+
+	r.currentRemoteDescription = &desc
+
+	r.remoteDescription = &sdp.SessionDescription{}
+
+	return r.remoteDescription.Unmarshal(desc.Sdp)
+}
+
+// RTCOfferOptions describes the options used to control the offer creation process
+type RTCOfferOptions struct {
+	VoiceActivityDetection bool
+	IceRestart             bool
```

Comment:
According to Go style all acronyms should be capitalized (SDP, ICE, etc) Thoughts?